### PR TITLE
Filter by domains

### PR DESCRIPTION
**Pull request by emord**
https://github.com/dimagi/commcare-hq/pull/13822/files#r86357306

@esoergel

### CODE REVIEW

File: corehq/apps/hqadmin/reporting/reports.py [hunk 1]
```diff
@@ -805,14 +807,14 @@ def get_submitted_users():
...
+def get_case_owner_filters(domains):
     result = {'terms': {}}
 
-    mobile_user_ids = list(get_user_ids(True))
+    mobile_user_ids = list(get_user_ids(True, domains))
 
     def all_groups():
-        for domain in Domain.get_all():
-            for group in Group.by_domain(domain.name):
+        for domain in domains:
```
Comment (emord) [inline corehq/apps/hqadmin/reporting/reports.py]: I think this was getting every group that existed in hq everytime someone loaded admin reports...

Comment (czue) [inline corehq/apps/hqadmin/reporting/reports.py]: ha, wow it does look that way. and every domain doc.

**Comment by emord** [comment]
Ok. I think I could spend a couple of days making this file use ES better, but I'll leave that for another day. I think it should be a little nicer now

**Comment by emord** [comment]
:white_check_mark:

**Comment by esoergel** [comment]
awesome, thanks for the cleanup!

**Pull request merged**